### PR TITLE
[GSProcessing] Support `:` in label column names

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* Previously, label column names that included colons in their name like `label:String`, which is how Neptune Analytics exports CSV columns, would trigger an SQL syntax error because of the way we defined the label DataFrame schema using an SQL string. 
 
For example with a column like `isFraud:Int`

```
[PARSE_SYNTAX_ERROR] Syntax error at or near ':'.(line 1, pos 3)
== SQL ==
idx: int, isFraud:Int: string
---^^^
```

* This PR programmatically defines the schema to avoid this issue.
* Modify a test to match the corner case.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_label_transformation.py [hunk 1]
```diff
@@ -75,10 +76,18 @@ def apply(self, input_df: DataFrame) -> DataFrame:
             *original_cols,
         )
 
+        # Create schema for label_df
```
Comment (jalencato) [inline graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_label_transformation.py:79]: What if the label column name is not like the format "idx:Int"?

Comment (thvasilo) [reply graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_label_transformation.py:79]: The `idx` column is one we create through the call to `list(enumerate(str_indexer_model.labelsArray[0]))`

**Review by jalencato — APPROVED** [review]
LGTM
